server/worker/api.go

```diff
@@ -29,7 +29,9 @@ func (api *Client) Get(endpoint string) ([]byte, error) {
 	client := &http.Client{}
 	res, err := client.Do(req)
 	if res.StatusCode != http.StatusOK {
-		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+		errorMessage := fmt.Sprintf("Request failed\nEndpoint: %s\nStatus: %s", endpoint, strconv.Itoa(res.StatusCode))
```

Comment:
FYI: I'd prefer keeping log messages mostly on single line. I'd consider new lines (with indentation to make following lines of the log record visually related to first log line) in cases when providing additional context (i.e. background information about the error or how to fix the error)